src/components/CustomCheckbox.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Checkbox, Tooltip } from 'antd';
+
+import { CheckboxState } from '../constants';
+import {
+  HighlightStar,
+  IndeterminateHighlightStar,
+  NoHighlightStar,
+} from './Icons';
+import { CheckboxDisplayOptions } from '../types';
+
+interface CustomCheckboxProps {
+  checkboxType: 'hide' | 'highlight';
+  status: CheckboxState;
+  clickHandler: () => void;
+}
+
+const CustomCheckbox: React.FunctionComponent<CustomCheckboxProps> = (
+  props: CustomCheckboxProps
+): JSX.Element => {
+  const { checkboxType, status, clickHandler } = props;
+
+  const getDisplayOptions = (): CheckboxDisplayOptions => {
+    switch (status) {
+      case CheckboxState.Checked:
+        return {
+          hideTooltipText: 'Hide',
+          highlightTooltipText: 'Highlight',
+          highlightAriaLabel: 'true',
+          highlightIcon: HighlightStar,
+        };
+      case CheckboxState.Unchecked:
+        return {
+          hideTooltipText: 'Show',
+          highlightTooltipText: 'Remove highlight',
+          highlightAriaLabel: 'false',
+          highlightIcon: NoHighlightStar,
+        };
+      case CheckboxState.Indeterminate:
+        return {
+          hideTooltipText: 'Show',
+          highlightTooltipText: 'Remove highlight',
+          highlightAriaLabel: 'mixed',
+          highlightIcon: IndeterminateHighlightStar,
+        };
+    }
+  };
+
+  const getDisplayElements = (): {
+    checkboxToRender: JSX.Element;
+    tooltipText: string;
+  } => {
+    const {
+      hideTooltipText,
+      highlightTooltipText,
+      highlightAriaLabel,
+      highlightIcon,
+    } = getDisplayOptions();
```

Comment:
Just a thought for readability and organizing the code:  since 3 out of the 4 options are for highlight and the other one is for hide, I wonder if you want to make this getDisplayOptions conditional on the checkboxType somehow, and then it can only return options specific to the type you're asking about.

Reply:
like maybe `getDisplayOptionsForHide() : HideDisplayOptions` and `getDisplayOptionsForHighlight(): HighlightDisplayOptions` ?

Reply:
Appreciate this comment because I actually went back and forth on this a lot!

Reply:
FYI, this fits well in the next change set so I'm going to merge this and update this alongside the highlight functionality